### PR TITLE
Refactor gateway flow

### PR DESCRIPTION
**Pull request by tobsan**
Basically changes the flow of how we create gateways, which means the way we work with the component tests had to be changed, so I improved the general component test structure while at it.

### CODE REVIEW

File: libsoftwarecontainer/src/gateway/dbus/dbusgatewayinstance.cpp [hunk 1]
```diff
@@ -22,12 +22,18 @@
...
+// These lines are needed in order to define the fields, which otherwise would
+// yield linker errors.
+constexpr const char *DBusGatewayInstance::SESSION_CONFIG;
+constexpr const char *DBusGatewayInstance::SYSTEM_CONFIG;
+
 DBusGatewayInstance::DBusGatewayInstance(ProxyType type
                        , const std::string &gatewayDir
-                       , const std::string &name)
-    : Gateway(ID)
+                       , std::shared_ptr<ContainerAbstractInterface> container)
```
Comment (breakreturn) [inline libsoftwarecontainer/src/gateway/dbus/dbusgatewayinstance.cpp]: This sticks out compared to the other code, the commas and colon etc.

File: libsoftwarecontainer/src/gateway/files/filegateway.cpp [hunk 2]
```diff
@@ -21,8 +21,8 @@
 
 namespace softwarecontainer {
 
-FileGateway::FileGateway()
-    : Gateway(ID)
+FileGateway::FileGateway(std::shared_ptr<ContainerAbstractInterface> container)
+    : Gateway(ID, container)
```
Comment (breakreturn) [inline libsoftwarecontainer/src/gateway/files/filegateway.cpp]: Style

File: libsoftwarecontainer/src/softwarecontainer.cpp [hunk 3]
```diff
@@ -117,10 +117,13 @@ bool SoftwareContainer::init()
 
 #ifdef ENABLE_NETWORKGATEWAY
     try {
-        addGateway(new NetworkGateway(m_containerID,
-                                      m_config->bridgeDevice(),
-                                      m_config->bridgeIPAddress(),
-                                      m_config->bridgeNetmaskBitLength()));
+        m_gateways.push_back( std::unique_ptr<Gateway>( new NetworkGateway(
```
Comment (breakreturn) [inline libsoftwarecontainer/src/softwarecontainer.cpp]: Too much spaces after parentheses.

**Review by breakreturn — CHANGES_REQUESTED** [review]
Minor style comments

**Comment by tobsan** [comment]
Updated after your comments @breakreturn

**Review by breakreturn — APPROVED** [review]

**Review by DunderRoffe — APPROVED** [review]

**Pull request merged**